cartographer/cloud/internal/map_builder_server.cc

```diff
@@ -51,8 +51,11 @@ const common::Duration kPopTimeout = common::FromMilliseconds(100);
 
 MapBuilderServer::MapBuilderServer(
     const proto::MapBuilderServerOptions& map_builder_server_options,
-    std::unique_ptr<mapping::MapBuilderInterface> map_builder)
-    : map_builder_(std::move(map_builder)) {
+    MapBuilderServer::MapBuilderFactory map_builder_factory)
```

Comment:
This pattern is also necessary for cartographer_ros Node and MapBuilderBridge. Is there an alternative to specify the global SLAM callback later?